bin/partitions.toit

```diff
@@ -0,0 +1,255 @@
+            --default=true,
+        cli.OptionEnum "state" ["new", "pending-verify", "valid", "aborted", "undefined"]
+            --short_help="The new state of the partition."
+            --default="valid",
+        cli.OptionInt "select-entry"
+            --short_help="The select entry to update. (default: 0 for ota_0, 1 for ota_1)"
+      ]
+      --rest=[
+        cli.Option "partition"
+            --short_help="Partition to set as boot."
+            --required,
+      ]
+      --run=:: set_ota_state it
+  cmd.add set_ota_state_cmd
+
+  version_cmd := cli.Command "version"
+      --short_help="Print the version and exit."
+      --run=:: print_version
+
+  cmd.run args
+
+with_esptool parsed/cli.Parsed [block]:
+  esptool_path := parsed["esptool"]
+  port := parsed["port"]
+  partition_table_offset_str := parsed["partition-table-offset"]
+  partition_table_size_str := parsed["partition-table-size"]
+
+  partition_table_offset := partition_table_offset_str.starts_with "0x"
+      ? int.parse partition_table_offset_str[2..] --radix=16
+      : int.parse partition_table_offset_str
+  partition_table_size := partition_table_size_str.starts_with "0x"
+      ? int.parse partition_table_size_str[2..] --radix=16
+      : int.parse partition_table_size_str
+
+  esptool := Esptool esptool_path
+      --port=port
+      --partition_table_offset=partition_table_offset
+      --partition_table_size=partition_table_size
+
+  block.call esptool
+
+print_partition_table parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool |
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    print "# Name, Type, SubType, Offset, Size, Size in K"
+    table.do: | partition/Partition |
+      type_string/string := ?
+      if partition.type == 0: type_string = "app"
+      else if partition.type == 1: type_string = "data"
+      else: type_string = "$partition.type"
+
+      k_size := partition.size / 1024
+      print "$partition.name, $type_string, $partition.subtype, 0x$(%x partition.offset), 0x$(%x partition.size), $(k_size)K"
+
+print_otadata parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool |
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    otadata_partition := table.find_otadata
+    otadata_offset := otadata_partition.offset
+    otadata_size := otadata_partition.size
+    otadata_bytes := esptool.read_flash
+        --offset=otadata_offset
+        --size=otadata_size
+
+    otadata := Otadata.decode otadata_bytes
+    2.repeat:
+      entry/SelectEntry := otadata.select_entries[it]
+      print """
+      otadata$it:
+        sequence-number: $entry.sequence_number
+        label: $entry.label.to_string_non_throwing
+        state: $entry.state ($(SelectEntry.state_stringify entry.state))
+        crc: $(%x entry.crc)"""
+      if it == 0: print
```

Comment:
This `it` is a bit too far from its block start.

Reply:
done.